server/routes/query_assist/routes.ts

```diff
@@ -83,11 +83,15 @@ export function registerQueryAssistRoutes(router: IRouter) {
         if (
           isResponseError(error) &&
           error.statusCode === 400 &&
-          error.body.includes(ERROR_DETAILS.GUARDRAILS_TRIGGERED)
+          // on opensearch >= 2.17, error.body is an object https://github.com/opensearch-project/ml-commons/pull/2858
+          JSON.stringify(error.body).includes(ERROR_DETAILS.GUARDRAILS_TRIGGERED)
```

Comment:
Is this route used anywhere when MDS is enabled?

Reply:
if MDS is enabled, log explorer is disabled, and this route won't be used by UI

Reply:
added a commit to disable it if data source is enabled